packages/@expo/cli/src/utils/codesigning.ts

```diff
@@ -0,0 +1,298 @@
+import {
+  convertCertificatePEMToCertificate,
+  convertKeyPairToPEM,
+  convertCSRToCSRPEM,
+  generateKeyPair,
+  generateCSR,
+  convertPrivateKeyPEMToPrivateKey,
+  validateSelfSignedCertificate,
+  signStringRSASHA256AndVerify,
+} from '@expo/code-signing-certificates';
+import { ExpoConfig } from '@expo/config';
+import { promises as fs } from 'fs';
+import { pki as PKI } from 'node-forge';
+import path from 'path';
+import { Dictionary, parseDictionary } from 'structured-headers';
+
+import { getExpoGoIntermediateCertificateAsync } from '../api/getExpoGoIntermediateCertificate';
+import { getProjectDevelopmentCertificateAsync } from '../api/getProjectDevelopmentCertificate';
+import { APISettings } from '../api/settings';
+import * as Log from '../log';
+import { createTemporaryProjectFile } from '../start/project/dotExpo';
+
+export type CodeSigningInfo = {
+  privateKey: string;
+  certificateForPrivateKey: string;
+  certificateChainForResponse: string[]; // empty means no need to serve the certificate chain in the multipart response
+};
+
+type StoredDevelopmentExpoRootCodeSigningInfo = {
+  easProjectId: string | null;
+  privateKey: string | null;
+  certificateChain: string[] | null;
+};
+const DEVELOPMENT_CODE_SIGNING_SETTINGS_FILE_NAME = 'development-code-signing-settings.json';
+const DevelopmentCodeSigningInfoFile =
+  createTemporaryProjectFile<StoredDevelopmentExpoRootCodeSigningInfo>(
+    DEVELOPMENT_CODE_SIGNING_SETTINGS_FILE_NAME,
+    {
+      easProjectId: null,
+      privateKey: null,
+      certificateChain: null,
+    }
+  );
```

Comment:
Logging out should arguably delete this file since the signing certificate and key are capabilities owned by the logged-in account.

Reply:
The issue is that logout doesn't occur within any particular project directory and we don't track project directories for the currently-logged-in user. I guess we could discuss putting the certificates somewhere else though that is independent of the project yet associated with the project somehow. I'll look into this.